command/server/tls_util.go

```diff
@@ -0,0 +1,178 @@
+package server
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/sha256"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"net"
+	"os"
+	"strings"
+	"time"
+)
+
+type CaCert struct {
+	PEM      string
+	Template *x509.Certificate
+	Signer   crypto.Signer
+}
+
+// GenerateCert creates a new leaf cert from provided CA template and signer
+func GenerateCert(caCertTemplate *x509.Certificate, caSigner crypto.Signer) (string, string, error) {
+	// Create the private key
+	signer, keyPEM, err := privateKey()
+	if err != nil {
+		return "", "", err
+	}
+
+	// The serial number for the cert
+	sn, err := serialNumber()
+	if err != nil {
+		return "", "", err
+	}
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		return "", "", err
+	}
+
+	if hostname == "" {
+		hostname = "localhost"
+	}
+
+	// Create the leaf cert
+	template := x509.Certificate{
+		SerialNumber:          sn,
+		Subject:               pkix.Name{CommonName: hostname},
+		BasicConstraintsValid: true,
+		KeyUsage:              x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth},
+		NotAfter:              time.Now().Add(365 * 24 * time.Hour),
+		NotBefore:             time.Now().Add(-1 * time.Minute),
+		IPAddresses:           []net.IP{net.ParseIP("127.0.0.1")},
```

Comment:
nit: I'd maybe add AKID (from the parent CA's SKID field) and add another SKID field if you're keeping them below? Just seems weird to add them to the CA but not the leaf.